sirepo/package_data/static/js/srw.js

```diff
@@ -820,7 +823,10 @@ SIREPO.app.controller('SourceController', function (appState, panelState, srwSer
     $scope.$on('modelChanged', function(e, name) {
         if (name == 'undulator' || name == 'tabulatedUndulator') {
             // make sure the electronBeam.drift is also updated
-            appState.saveQuietly('electronBeamPosition');
+            srwService.computeBeamParameters(() => {
```

Comment:
I'm concerned about this. The changes are sitting somewhere and then there's an async call which may never return (user closes laptop). The assumption you can just save a random set of changes "some time later" is not a good one. It seems to me that the saveChanges should be happening in the sendStatelessCompute callback, but I could be wrong. However, again, there's no synchronization with the model state at the time of the request and in the response.

Reply:
Not sure I follow here
```
if (replyHandler) {
    replyHandler();
}
```
occurs in the `requestSender.sendStatelessCompute` call. Do you mean something else by saveChanges "should be happening in sendStatelessCompute"?